File: Data Structures & Algorithms/linked-list-cycle-detection/submission-0.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */

class Solution {
    public boolean hasCycle(ListNode head) {
        Map<ListNode,ListNode> nextnodes = new HashMap<>();
        ListNode temp = head;
        while(temp != null){
            if(!nextnodes.containsKey(temp)){
                nextnodes.put(temp,temp.next);
            }else{
                return true;
            }
            temp = temp.next;
        }
       return false;
    }
}

File: Data Structures & Algorithms/remove-node-from-end-of-linked-list/submission-0.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next

class Solution:
    def removeNthFromEnd(self, head: Optional[ListNode], n: int) -> Optional[ListNode]:
        listlen = 0
        temp = head

        while temp:
            listlen += 1
            temp = temp.next
        
        i = 0
        temp = head
        prev = None
        while i < listlen-n:
            i += 1
            prev = temp
            temp = temp.next

        if prev == None:
            return head.next
        else:
             prev.next = temp.next
             return head

